### ISSUE TITLE
Import a large number of users (>10000) into GLPI 9.1-RC2

### ISSUE THREAD

**Issue by gloomy-master** [glpi-project/glpi#1024]
Question about importing users.

In the structure there are many domain OU - Organizational Unit inside which another OU, the depth reaches more than 6-7 levels. GLPI sees only a limited part of the users when importing if you specify "search base (baseDN)" at the top level.

If you specify a specific OU and the full path to it in the "search base (baseDN)" the user sees and starts to import.

**Comment by orthagh** [comment]
as i know, there isn't limit for OU traversal.

Some questions:
- Do you see users from above OU ?
- If you set your path on an intermdiate one, did you see your user ?
- Do you more than 1000 users ?

**Comment by gloomy-master** [comment]
When you import new users
Fatal error: Maximum execution time of 30 seconds exceeded in /usr/share/glpi/inc/authldap.class.php on line 1575

**Comment by wawax** [comment]
Hello,
You have to configure LDAP paged results in the advanced informations tab of the LDAP server.
By doing so, GLPi will perform the query, and get the results by paged (by default 100 results at the time).

**Comment by orthagh** [comment]
For "Synchronizing already imported users" we ship in the glpi tarball a script to achieve that 
See scripts/ldap_mass_sync.php 

You can execute it by system crontab with the parameter action: 
0: import users only
1: synchronize existing users only
2: import & synchronize users

**Comment by gloomy-master** [comment]
wawax, I changed the setting and did 300, 500, 1000
Problems remain and the error is gone.
Note that import about 1000 remaining accounts, but does not know what OU they stayed.

Noticed that the Synchronizing already imported users
available to synchronize all 1500 users, whereas users shows 11225

**Comment by wawax** [comment]
Please use the script as mentionned by orthagh.
The interface was not meant to add or synchronize such amount of users.
Otherwise you'll have to increase the max_execution_time in php.ini, but using the script is definitly the good way to go

**Comment by gloomy-master** [comment]
Question about synchronization, I understand why synchronization shows the changes are not all at once. I specify I changed the user's properties, outcome variables appear in GLPI after only 2 synchronization.

**Comment by gloomy-master** [comment]
Please answer to the last question and close this issue)

**Comment by tomolimo** [comment]
Hello,
Could you detail your question?
Thank you

**Comment by gloomy-master** [comment]
Hello, tomolimo.

I will try to describe in detail.

Example:
I have made changes in the properties of the fields in the user accounts in Active Directory, after you ran the script synchronization via ldap_mass_sync.php and noticed that during the first synchronization, the changes have required me to users in there and GLPI appeared after restarting the synchronization.

+ I noticed that the script about always loads the changes of about 300-400 users accounts
